source/opt/scalar_analysis.cpp

```diff
@@ -13,16 +13,41 @@
 // limitations under the License.
 
 #include "opt/scalar_analysis.h"
+
 #include <algorithm>
 #include <functional>
 #include <string>
 #include <utility>
 
 #include "opt/ir_context.h"
 
+// Transforms a given scalar operation instruction into a DAG representation.
+//
+// 1. Take an instruction and traverse the its operands until we reach a
+// constant node or any instruction which we do not know how to compute the
+// value of, such as a load.
```

Comment:
Try "or an instruction for which we do not know how to compute the value, such as a load."